pkg/controller/configmapcainjector/controller/configmap_injecting_controller.go

```diff
@@ -46,20 +46,21 @@ func (ic *configMapCABundleInjectionController) Sync(obj v1.Object) error {
 	if !api.HasInjectCABundleAnnotation(sharedConfigMap) {
 		return nil
 	}
-	// skip updating when the CA bundle is already there
-	if data, ok := sharedConfigMap.Data[api.InjectionDataKey]; ok && data == ic.ca {
-		return nil
-	}
+	// ensure data of configmap
+	return ic.ensureConfigMapCABundleInjection(sharedConfigMap)
+}
 
+// ensureConfigMapCABundleInjection will create or update configmap for the
+// CA bundle injection as appropriate.
+func (ic *configMapCABundleInjectionController) ensureConfigMapCABundleInjection(current *corev1.ConfigMap) error {
 	// make a copy to avoid mutating cache state
-	configMapCopy := sharedConfigMap.DeepCopy()
-
-	if configMapCopy.Data == nil {
-		configMapCopy.Data = map[string]string{}
+	configMapCopy := current.DeepCopy()
+	// skip updating when the CA bundle is already there
+	if data, ok := configMapCopy.Data[api.InjectionDataKey]; ok &&
+		data == ic.ca && len(configMapCopy.Data) == 1 {
+		return nil
 	}
-
-	configMapCopy.Data[api.InjectionDataKey] = ic.ca
-
-	_, err := ic.configMapClient.ConfigMaps(configMapCopy.Namespace).Update(configMapCopy)
+	configMapCopy.Data = map[string]string{api.InjectionDataKey: ic.ca}
```

Comment:
Do the deep copy before this line instead, you're not modifying anything in the code above.

Reply:
done